src/Microsoft.DocAsCode.Build.RestApi/Swagger/Internals/SwaggerJsonBuilder.cs

```diff
@@ -255,5 +289,16 @@ private static SwaggerObject ResolveSwaggerObject(SwaggerObjectBase swaggerObjec
 
             throw new ArgumentException($"When resolving reference for {nameof(SwaggerReferenceObject)}, only support {nameof(SwaggerObject)} and {nameof(SwaggerReferenceObject)} as parameter.");
         }
+
+        private static string GetDirectory(string path)
+        {
+            var directory = Path.GetDirectoryName(path);
+            if (string.IsNullOrEmpty(directory))
+            {
+                throw new DocfxException($"Directory of swagger file path {path} should not be null or empty.");
```

Comment:
"C:\s.json" is supported while "s.json" without directoty is not support. the path is passed by "Path.Combine(file.BaseDir, file.File)" in RestApiDocumentProcessor, so we assume that the directory exists here?

---
In reply to: [101675921](https://github.com/dotnet/docfx/pull/1308#discussion_r101675921) [](ancestors = 101675921)